llvm/include/llvm/Target/TargetSelectionDAG.td

```diff
@@ -1113,6 +1113,10 @@ def not  : PatFrag<(ops node:$in), (xor node:$in, -1)>;
 def vnot : PatFrag<(ops node:$in), (xor node:$in, immAllOnesV)>;
 def ineg : PatFrag<(ops node:$in), (sub 0, node:$in)>;
 
+def or_disjoint : PatFrag<(ops node:$x, node:$y), (or node:$x, node:$y), [{
```

Comment:
Please see the existing or_is_add and add_like in RISCVInstrInfo.td.  We can move into generic, but let's do that separately.